fun/middleware/enforce_legal_acceptance.py

```diff
@@ -3,72 +3,61 @@
 conditions are published"""
 
 import re
-from django.conf import settings
+
 from django.shortcuts import redirect
+
 from payment.models import legal_acceptance
 
-DEFAULT_AGREEMENT_WHITELIST = map(
+AGREEMENT_WHITELIST = map(
     re.compile,
     (
-        r"""^/payment/.*""",
-        r"""^/xblock/.*""",
-        r"""^/account/setting.*""",
-        r""".*accept.*""",
-        r"""^/(news|login|contact|about|help|legal|privacy|honor|charte)/?$""",
-        r""".*/login_ajax.*""",
-        r"""^/static.*""",
-        r""".*search.*""",
-        r""".*revision.*""",
-        r"""^/static/.*""",
-        r"""^/api/user/.*""",
-        r"""^/c4x.*""",
-        r""".*logout.*$""",
-        r""".*asset.*""",
-        r""".*\.(png|font|html|js|css|jpeg|pdf|gif|mp4).*""",
-        r"""^/?$""",
+        r'^/payment/.*',
+        r'^/xblock/.*',
+        r'^/account/setting.*',
+        r'.*accept.*',
+        r'^/(news|login|contact|about|help|legal|privacy|honor|charte)/?$',
+        r'.*/login_ajax.*',
+        r'^/static.*',
+        r'.*search.*',
+        r'.*revision.*',
+        r'^/static/.*',
+        r'^/api/user/.*',
+        r'^/c4x.*',
+        r'.*logout.*$',
+        r'.*asset.*',
+        r'.*\.(png|font|html|js|css|jpeg|pdf|gif|mp4).*',
+        r'^/?$',
     )
 )
-def get(self, key, default):
-    return getattr(self, key) if hasattr(self, key) else default
 
-LMS_BASE = get(settings, "LMS_BASE", "")
-PROT = get(settings, "HTTPS", "off") == "on" and "s" or ""
-LMS_REDIR = LMS_BASE and ("http%s://%s/" % (PROT, LMS_BASE)) or ""
-DEFAULT_AGREEMENT_FORM = "%s/payment/terms/" % (LMS_REDIR)
+TERMS_AND_CONDITIONS_AGREEMENT = '/payment/terms/'
+
 
 def terms_accepted(func):
```

Comment:
I think the name of this wrapper is confusing, what about renaming it: `ensure_accepted_terms`? or something alike.

Reply:
Yes I thought the same but didn't take action. Let's do it.